core/src/main/scala/cats/TraverseEmpty.scala

```diff
@@ -0,0 +1,26 @@
+package cats
+
+import simulacrum.typeclass
+
+/**
+ * `TraverseEmpty`, also known as `Witherable`, represents list-like structures
+ * that can essentially have a `traverse` and a `filter` applied as a single
+ * combined operation (`traverseFilter`).
+ *
+ * Based on Haskell's [[https://hackage.haskell.org/package/witherable-0.1.3.3/docs/Data-Witherable.html Data.Witherable]]
+ */
+
+@typeclass
+trait TraverseEmpty[F[_]] extends FunctorEmpty[F] {
+  def traverse: Traverse[F]
+
+  override def functor: Functor[F] = traverse
```

Comment:
I haven't spent a lot of time on the this type of encoding.  Is there a reason that `functor` should ever _not_ be the `traverse`?  In other words, do we lose anything by making this final?

Reply:
Nope, I agree fully, I'll change it :)